GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/TestRailManager/TestRailClientWrapper.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Net;
+using System.Reflection;
+using GherkinSyncTool.Exceptions;
+using GherkinSyncTool.Synchronizers.TestRailSynchronizer.TestRailManager.Model;
+using Newtonsoft.Json.Linq;
+using NLog;
+using TestRail;
+using TestRail.Types;
+using TestRail.Utils;
+
+namespace GherkinSyncTool.Synchronizers.TestRailSynchronizer.TestRailManager
+{
+    public class TestRailClientWrapper
+    {
+        private static readonly Logger Log = LogManager.GetLogger(MethodBase.GetCurrentMethod()?.DeclaringType?.Name);
+        private readonly TestRailClient _testRailClient;
+
+        public TestRailClientWrapper(TestRailClient testRailClient)
+        {
+            _testRailClient = testRailClient;
+        }
+
+        public Case AddCase(CreateCaseRequest createCaseRequest)
+        {
+            var addCaseResponse =
+                _testRailClient.AddCase(createCaseRequest.SectionId, createCaseRequest.Title, createCaseRequest.TypeId,
+                    createCaseRequest.PriorityId, createCaseRequest.Estimate, createCaseRequest.MilestoneId,
+                    createCaseRequest.Refs, JObject.FromObject(createCaseRequest.CustomFields), createCaseRequest.TemplateId);
+
+            ValidateRequestResult(addCaseResponse);
+
+            Log.Info($"Created: [{addCaseResponse.Payload.Id}] {addCaseResponse.Payload.Title}");
+            return addCaseResponse.Payload;
+        }
+
+        public void UpdateCase(UpdateCaseRequest createCaseRequest)
+        {
+            var testRailCase = GetCase(createCaseRequest.CaseId);
+
+            //TODO: handle with test case content (not only title) 
```

Comment:
Let's add subtasks in JIRA for each TODO comments so we will not forget to fix them.

Reply:
Ok